### ISSUE TITLE
API Calls

### ISSUE THREAD

**Issue by kyuksel** [DataBiosphere/leonardo#681]
Please see the [Performance Testing tech doc](https://docs.google.com/document/d/1b1baxAwaNMy71nLr3VZggQ6jo-UiCIk_GtpccdmYtuI) for context.

**Victory condition:** We chained the API calls listed below with configurable pauses in between, all of which can be executed by a configurable number of users concurrently.

We will start with defining a [Gatling scenario](https://gatling.io/docs/2.3/general/scenario/) that chains the following [exec](https://gatling.io/docs/2.3/general/scenario/#exec)’s and can be
executed in parallel for a configurable number of users. For an example, see
[serviceAccountsTest.scala](serviceAccountsTest.scala). The API calls we’ll want to make, to mimic what the UI does, include the following:

1. [create cluster](https://github.com/broadinstitute/mcnulty/blob/aj-ashg-workshop/src/test/scala/simulations/CreateClusterSimulation.scala) (use the v2 API - /api/cluster/v2/{googleProject}/{clusterName})
2. [list clusters by project](https://github.com/broadinstitute/mcnulty/blob/3ab310ea38e040eab39836ecc6de234cd0e57e4c/src/test/scala/simulations/serviceAccountsTest.scala#L2655): call to GET /api/clusters/{googleProject} every 30 seconds (but we should make the frequency configurable), which is what the FireCloud UI does until the cluster is created or it errors. Make sure to capture any errors. Errors could be captured in and referred to later by adding them as key-value pairs to the [Gatling session](https://gatling.io/docs/2.3/session/session_api/#concept), an example of which can be seen [here](https://github.com/broadinstitute/mcnulty/blob/3ab310ea38e040eab39836ecc6de234cd0e57e4c/src/test/scala/simulations/serviceAccountsTest.scala#L2613). 
3. [get cluster](https://leonardo.dsde-perf.broadinstitute.org/#!/cluster/getCluster)
4. [patch cluster](https://github.com/DataBiosphere/leonardo/pull/669)
5. [localize notebook](https://github.com/broadinstitute/mcnulty/blob/3ab310ea38e040eab39836ecc6de234cd0e57e4c/src/test/scala/simulations/LocalizeAndOpenNotebookSimulation.scala#L64) (we can start with the [ASHG notebook](https://storage.cloud.google.com/firecloud-workshops/181017-ashg18/notebooks/cluster_analysis.ipynb?folder=true&organizationId=548622027621&_ga=2.53241512.-1060415104.1522950615))
6. [setCookie](https://github.com/broadinstitute/mcnulty/blob/3ab310ea38e040eab39836ecc6de234cd0e57e4c/src/test/scala/simulations/LocalizeAndOpenNotebookSimulation.scala#L77)
7. [open notebook](https://github.com/broadinstitute/mcnulty/blob/3ab310ea38e040eab39836ecc6de234cd0e57e4c/src/test/scala/simulations/LocalizeAndOpenNotebookSimulation.scala#L89)

While implementing those chains, we will keep an eye towards making them more modular later on, if need be (e.g. an individual test for each API call).

**Comment by MatthewBemis** [comment]
Closing in favor of newly created performance testing tickets.
